### PR TITLE
Feature/person overview

### PR DESCRIPTION
**Pull request by azurebrd**
Linting with 'lein eastwood'
There are 2 warnings from gene_test.clj and 1 warning from test_routing.clj, which I didn't modify, so I don't know whether warnings are okay to ignore.
There are 2 warnings from publications.clj about def-in-def. I don't know why it's bad, nor alternatively, how to get around doing group-by without defining results into a new variable.

### CODE REVIEW

**Comment by sibyl229** [comment]
Hi @azurebrd def-in-def is probably not what you want, because `def` is global to the namespace, maybe what you need is `let`? 

I haven't tried eastwood lately to know if the other warnings are okay.

I think the code is nice. I'm happy with everything else. Just some indentations are off, can you try have your text editor reindent everything?

**Comment by azurebrd** [comment]
Thanks @sibyl229 ! I'll try out let tomorrow, and I remember now Adam telling me about vim-fireplace for the indenting.

File: src/rest_api/classes/person/widgets/publications.clj [hunk 1]
```diff
@@ -0,0 +1,71 @@
+(ns rest-api.classes.person.widgets.publications
+  (:require
+   [clojure.string :as str]
+   [datomic.api :as d]
+   [pseudoace.utils :as pace-utils]))
+
+
+(defn publications [person]
+   (let [db (d/entity-db person)
+         dataArrayTuples
```
Comment (mgrbyte) [inline src/rest_api/classes/person/widgets/publications.clj]: Suggest renaming `dataArrrayTuples` to just `data` or `datoms`

File: src/rest_api/classes/person/widgets/publications.clj [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                       :object
+                        (pace-utils/vmap
+                         :taxonomy "all"
+                         :class "paper"
+                         :label (:paper/brief-citation paper)
+                         :id (:paper/id paper))
+                       :brief_citation (:paper/brief-citation paper)
+                     ))))
+              (seq))]
+     (def data (group-by :year dataArrayTuples))
```
Comment (mgrbyte) [inline src/rest_api/classes/person/widgets/publications.clj]: You can eliminate the `eastwood` warnings by removing this line (32),
and adding `((group-by :year))` as the final step in the `->>` pipeline, just below `(seq)` above.

File: src/rest_api/classes/person/widgets/publications.clj [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
```
Comment (mgrbyte) [inline src/rest_api/classes/person/widgets/publications.clj]: Same as above w/regard to naming and eliminating `lein eastwood` warnings.

**Review by mgrbyte — COMMENTED** [review]
Generally, logic looks good.
Style-wise, we're trying to keep to "standard" clojure conventions:
  naming: var-names-like-this
  spacing: 2 space indent
  braces brackets wrap up (never on their own line)

File: src/rest_api/classes/person/widgets/laboratory.clj [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                                   :id (:person/id rep)}) ))
+                       )
+                     )))
+              (seq))]
+     {:data (if (empty? data) nil data)
+   :description
+   "allele designation of the affiliated laboratory"}))
+;; note that the perl catalyst interface does not return proper data for "rep", it gives a single value while the data could have multiple persons, e.g. WBPerson8705
+
+(defn gene-classes [person]
```
Comment (mgrbyte) [inline src/rest_api/classes/person/widgets/laboratory.clj]: Could this be re-worked so just one datomic query is made?
Currently: 1 + number-of-labs-per-person queries.

Something like:
```clojure
(d/q '[:find [?geneclass ...]
         :where
         [?lab :laboratory/registered-lab-members ?person]
         [?geneclass :gene-class/designating-laboratory ?lab]]
        db (:db/id person))
```

**Comment by azurebrd** [comment]
Thanks @mgrbyte !  

The (seq)(group-by :year) worked great.

For reducing the number of gene class queries, it makes sense to query for all the gene classes of a given person, and follow them to the gene class's laboratory, but to get the :lab :label/:id is there a way to follow the gene_class without making a new query into each gene_class's laboratory ?  Something like this doesn't work

                                (let [geneclass (d/entity db oid)]
                                  (pace-utils/vmap
                                   :lab
                                      {:taxonomy "all"
                                       :class "laboratory"
                                       :label (:gene-class/designating-laboratory geneclass)

because it's the :ref entity instead of the :laboratory/id

In ACeDB gene_class xrefs many-to-one to laboratory, and laboratory xrefs uniquely to gene_class, so I've switched the order of the querying, gene_class first then laboratory within that, but that doesn't reduce the number of queries, just sort of returns the data in a more pair-like way.

If you meant that we can query for a person's laboratory and gene_class separately and aggregate them, a person can have multiple labs, and each gene_class has to be associated to its own laboratory.  If that weren't the case and we could aggregate the data off of separate queries, I wouldn't know how to write the syntax, but it'd be nice to know that it would be okay to do things that way.  I've been following the format of existing queries to generate the data, but there's been times that I've wanted to query things separately and aggregate afterward.

File: src/rest_api/classes/person/widgets/publications.clj [hunk 5]
```diff
@@ -57,8 +56,7 @@
                          :id (:paper/id paper))
                        :brief_citation (:paper/brief-citation paper)
                      ))))
-              (seq))]
-     (def data (group-by :year dataArrayTuples))
+              (seq)(group-by :year))]
```
Comment (mgrbyte) [inline src/rest_api/classes/person/widgets/publications.clj]: Nearly! 
Think you want:
```
...
(group-by :year)
(seq))
````

File: src/rest_api/classes/person/widgets/laboratory.clj [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                         :id (:laboratory/id laboratory))
+                        :allele (:laboratory/allele-designation laboratory)
+                        :rep
+                        (->> (:laboratory/representative laboratory)
+                             (map (fn [rep]
+                                  {:taxonomy "all"
+                                   :class "person"
+                                   :label (:person/standard-name rep)
+                                   :id (:person/id rep)}) ))
+                       )
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/laboratory.clj]: You should pull these braces up

File: src/rest_api/classes/person/widgets/laboratory.clj [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                     )))
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/laboratory.clj]: You should pull these braces up

File: src/rest_api/classes/person/widgets/laboratory.clj [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                                      :class "laboratory"
+                                      :label (:laboratory/id laboratory)
+                                      :id (:laboratory/id laboratory)}
+                                  :desc (first (:gene-class/description geneclass))
+                                  :gene_class
+                                              {:taxonomy "all"
+                                               :class "gene_class"
+                                               :label (:gene-class/id geneclass)
+                                               :id (:gene-class/id geneclass)})
+                        ) )))
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/laboratory.clj]: You should pull these braces up

File: src/rest_api/classes/person/widgets/laboratory.clj [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                              (first (->> (:laboratory/representative old-laboratory)
+                                   (map (fn [rep]
+                                        {:taxonomy "all"
+                                         :class "person"
+                                         :label (:person/standard-name rep)
+                                         :id (:person/id rep)}) )) )
+                              "no representative")
+                         )
+                       ))
+                     )))
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/laboratory.clj]: You should pull these braces up

File: src/rest_api/classes/person/widgets/overview.clj [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+                       :institution (:address/institution old-address)
+                       :street-address (:address/street-address old-address)
+                       :country (:address/country old-address)
+                       :main-phone (:address/main-phone old-address)
+                       :lab-phone (:address/lab-phone old-address)
+                       :office-phone (:address/office-phone old-address)
+                       :other-phone (:address/other-phone old-address)
+                       :fax (:address/fax old-address)
+                       :web-page (:address/web-page old-address)
+                      )
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/overview.clj]: You should pull these braces up

File: src/rest_api/classes/person/widgets/publications.clj [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+     {:data (if (empty? data) nil data)
+   :description
+   "meeting abstract publications by this person."}))
+
+
+(def widget
+  { :publications             publications
+    :meeting_abstracts        meeting-abstracts
+  })
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/publications.clj]: You should pull these braces up

File: src/rest_api/classes/person/widgets/tracking.clj [hunk 12]
```diff
@@ -0,0 +1,34 @@
+(ns rest-api.classes.person.widgets.tracking
+  (:require
+   [datomic.api :as d]
+   [rest-api.formatters.date :as date]))
+
```
Comment (adamjohnwright) [inline src/rest_api/classes/person/widgets/tracking.clj]: Should remove extra lines like this one

**Review by adamjohnwright — CHANGES_REQUESTED** [review]

**Comment by azurebrd** [comment]
@a8wright Went through formatting with vim-fireplace and cleaned it up.

**Review by adamjohnwright — APPROVED** [review]

**Comment by adamjohnwright** [comment]
you will need to do a "git merge develop" and then fix the merge conflicts. Then do a merge commit. and push. I think it will be ready to pull at that point!  Great work!!

**Comment by azurebrd** [comment]
@a8wright Thanks Adam !  I think it's done.  I ended up in the development branch, where I merged, fixed, and pushed it.  I then came back to this branch, merged in develop, saw the changes and it all runs well, so I think it's good.  I think I just didn't 'git pull' before merging develop, which is why it didn't have any changes in develop to merge.  Sorry about that.

**Comment by adamjohnwright** [comment]
@azurbrd No problem. I use the gitflow tool and it doesn't pull from develop for me. I thought the "git hf update" would do it but no such luck lately. I am glad you are able to get it merged.

**Comment by azurebrd** [comment]
Thanks !

**Pull request merged**